thingif/src/main/java/com/kii/thingif/command/Action.java

```diff
@@ -30,5 +24,12 @@
  *
  */
 public interface Action {
+    /**
```

Comment:
@shinsuke-nara 
> The problem is who should validate value of member fields.

I think this is not a problem. 

The way of use abstract class for Action like:

```java
public abstract class Action {

     @NonNull public final String name
     Action(String name) {
        if (checkEmpty(name)) {
            throw IllegalArgumentException()
        }
        this.name = name
    }

    public String getActionName() {
        return this.name
    }
}
```
It is not simple to use. From developer, need to implement like this 

```java
class TurnPower extends Action {
    public Boolean power;

    // name is decide runtime
    public TurnPower(String name, Boolean power) {
        super(name);
        this.power = power;
    }

    // name is decide on compile time, if the field is string, can not use this way :(
    public TurnPower(Boolean power) {
        super("turnPower);
        this.power = power;
    }
}
```
So as a result, we left 2 ways to developer. I think less choices is simple.